core/drivers/stm32_rng.c

```diff
@@ -145,6 +160,9 @@ static void gate_rng(bool enable, struct stm32_rng_instance *dev)
 	}
 
 	may_spin_unlock(&dev->lock, exceptions);
+
+	if (enable)
+		warmup(io_pa_or_va(&dev->base));
```

Comment:
It looks a bit inefficient to wait here, you could instead continue to run useful code and defer the wait until some random value is requested, no?
Or even simpler: why not simply increase `RNG_TIMEOUT_US`?

Reply:
> Or even simpler: why not simply increase RNG_TIMEOUT_US?

maybe yes!
I will update...